src/natcap/invest/validation.py

```diff
@@ -156,12 +121,12 @@ def get_sufficient_keys(args):
     return sufficient_keys
 
 
-def check_directory(dirpath, exists=False, permissions='rx'):
+def check_directory(dirpath, must_exist=True, permissions='rx', **kwargs):
```

Comment:
I couldn't determine what purpose the `**kwargs` serve here?

Reply:
I see this used in most the functions here, so I won't comment on them all. I'm sure it's a discussion I missed out on, but am curious the use behind it!

Reply:
This is because now the arg's spec (`ARGS_SPEC["args"][arg_name]`) is passed directly into the type validation function. I like this change because it's more straightforward this way, but we need to allow the possibility of extra spec properties in a user-defined spec, hence the `**kwargs` to absorb them.
https://github.com/emlys/invest/blob/542a7c39c6ce06f8d4a50b662dc5ea6ea0507bb5/src/natcap/invest/validation.py#L894